src/classes/CRLP_Batch_Base_NonSkew.cls

```diff
@@ -298,6 +304,31 @@ public abstract class CRLP_Batch_Base_NonSkew extends CRLP_Batch_Base {
             );
             insert logEntry;
         }
+
+        // If chunk mode is enabled and we (probably) have more records to process,
+        // submit another batchable job
+        if (isChunkModeEnabled && batchIteration > 1) {
```

Comment:
There's a train of thought on handling this a bit differently. Curious what you think. Rather than just restart the batch for the next chunk if more than one child batch was processed, the finish() method would query for the next Id to determine if there is one.

`SELECT Count() FROM Contact WHERE Id > :lastIdProcessed LIMIT 1`

If that returns back a number > 0, then the job is resubmitted. Otherwise it's not. The thought is that this would avoid queuing a new job that will not select any records. In the end the overall work still finishes at the same time, but this avoids the extra job by sacrificing a query. 

My thought is that it might work. `Id` is indexed, and the `LIMIT 1` clause will force the query to be quick.

What do you think?